src/libcrun/linux.c

```diff
@@ -1840,6 +1840,38 @@ make_parent_mount_private (const char *rootfs, libcrun_error_t *err)
   return 0;
 }
 
+// libcrun_create_kvm_device: explicitly adds kvm device.
+int
+libcrun_create_kvm_device (libcrun_container_t *container, libcrun_error_t *err)
+{
+  int ret, rootfsfd;
+  struct device_s kvm_device = { "/dev/kvm", "c", 10, 232, 0666, 0, 0 };
+  cleanup_close int devfd = -1;
+  cleanup_close int rootfsfd_cleanup = -1;
+  const char *rootfs = get_private_data (container)->rootfs;
+  if (rootfs == NULL)
+    rootfsfd = AT_FDCWD;
+  else
+    {
+      rootfsfd = rootfsfd_cleanup = open (rootfs, O_PATH);
+      if (UNLIKELY (rootfsfd < 0))
+        return crun_make_error (err, errno, "open `%s`", rootfs);
+    }
+  get_private_data (container)->rootfsfd = rootfsfd;
+  devfd = openat (rootfsfd, "dev", O_RDONLY | O_DIRECTORY);
+  if (UNLIKELY (devfd < 0))
+    return crun_make_error (err, errno, "open /dev directory in `%s`", rootfs);
+  int is_user_ns = 0;
+  is_user_ns = (get_private_data (container)->unshare_flags & CLONE_NEWUSER);
+
+  ret = create_dev (container, devfd, &kvm_device, is_user_ns ? true : false, true, err);
+
```

Comment:
please drop the empty line between `ret = ...` and the `if (...`

Reply:
Sure.